llmfoundry/tokenizers/tiktoken.py

```diff
@@ -101,6 +112,24 @@ def get_vocab(self) -> Dict[str, int]:
             except KeyError:
                 pass
 
+        # As far as I can tell, we don't require get_vocab to completely work,
+        # but when using additional_special_tokens, Hugging Face determines the next
+        # token index to add with len(self.get_vocab()) so we need the _size_ of this dictionary to be correct.
+        extra_id_index = 0
+        candidate_extra_id = f'<extra_id_{extra_id_index}>'
+        indices_to_fill_in = {i for i in range(self.vocab_size)} - set(
+            vocab.values())
+
+        # Add enough indices to make get_vocab() the right length
+        for index_to_add in indices_to_fill_in:
+            # Make sure we don't overwrite a token that already exists
+            while candidate_extra_id in vocab:
+                extra_id_index += 1
+                candidate_extra_id = f'<extra_id_{extra_id_index}>'
```

Comment:
So there's a big vocab gap in the public tiktoken between a bunch of the special tokens and the ENDOFPROMPT token which is likely where the private special tokens are used
https://github.com/openai/tiktoken/blob/main/tiktoken_ext/openai_public.py#L72 (Pointed out by @samhavens ) - should we add to the end of this by appending or overwrite the intermediate tokens in that gap?

Reply:
I chose to just go off the end (after ENDOFPROMPT) because it seemed more straightforward and with no risk 🤷‍♂️ open to a different opinion though.

Reply:
sgtm